src/arch/x86/reg/mod.rs

```diff
@@ -1,9 +1,90 @@
 //! `GdbRegister` structs for x86 architectures.
 
+use core::convert::{TryFrom, TryInto};
+
 mod core64;
+mod core32;
 
 pub use core64::X86_64CoreRegs;
-pub use core64::X87FpuInternalRegs;
+pub use core32::X86CoreRegs;
 
 /// 80-bit floating point value
 pub type F80 = [u8; 10];
+
+/// FPU registers
+#[derive(Default)]
+pub struct X87FpuInternalRegs {
+    /// Floating-point control register
+    pub fctrl: u32,
+    /// Floating-point status register
+    pub fstat: u32,
+    /// Tag word
+    pub ftag: u32,
+    /// FPU instruction pointer segment
+    pub fiseg: u32,
+    /// FPU intstruction pointer offset
+    pub fioff: u32,
+    /// FPU operand segment
+    pub foseg: u32,
+    /// FPU operand offset
+    pub fooff: u32,
+    /// Floating-point opcode
+    pub fop: u32,
+}
+
+impl TryFrom<&[u8]> for X87FpuInternalRegs {
```

Comment:
Instead of implementing `TryFrom<&[u8]>` and `X87FpuInternalRegs::write`, `X87FpuInternalRegs` should just implement `Registers`. After all, the `try_from` method is essentially just `gdb_deserialize()`, and the `write` method is just `gdb_serialize()` :sweat_smile: